### ISSUE TITLE
Build Error: Unable to build gem nokogiri from official ruby:onbuild image

### ISSUE THREAD

**Issue by lvangool** [docker/docker#18578]
Description of problem:
Using a basic app that includes the nokogiri gem fails. The exact issue is reproducible on some servers (including this one). However it is not always a problem and the hypothesis is that this is related to using the overlay storage driver. This was raised in the nokogiri repo, but is more like a docker issue - see here: https://github.com/sparklemotion/nokogiri/issues/1370.

$ uname -a
```
Linux build1 3.19.0-39-generic #44~14.04.1-Ubuntu SMP Wed Dec 2 10:00:35 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

$ docker version
```
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64
```

$ docker info
```
Containers: 5
Images: 5398
Server Version: 1.9.1
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-39-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 16
Total Memory: 29.44 GiB
Name: banana
ID: QUOP:KIPQ:W3EP:32ZJ:T6TE:UFEU:VVN4:***
Debug mode (server): true
 File Descriptors: 12
 Goroutines: 21
 System Time: 2015-12-10T20:19:18.418550022Z
 EventsListeners: 0
 Init SHA1: 4fc7c03a06c675f115b39958c13486d53df10a14
 Init Path: /usr/lib/docker/dockerinit
 Docker Root Dir: /var/lib/docker
```

Environment: AWS server.

To reproduce, build using the following Dockerfile for a Ruby app that includes the gem nokogiri.
Step1: Create the following files: 

*Gemfile*
```
source 'https://rubygems.org'
gem 'nokogiri'
```
*Gemfile.lock*
```
GEM
  remote: https://rubygems.org/
  specs:
    mini_portile2 (2.0.0)
    nokogiri (1.6.7)
      mini_portile2 (~> 2.0.0.rc2)

PLATFORMS
  ruby

DEPENDENCIES
  nokogiri
```
*Dockerfile*
```
FROM ruby:2.2.3-onbuild
```

Step 2: Run `docker build .`

Output:
```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /usr/local/bundle/gems/nokogiri-1.6.7/ext/nokogiri
/usr/local/bin/ruby -r ./siteconf20151210-5-1yusha0.rb extconf.rb
checking if the C compiler accepts ... yes
Building nokogiri using packaged libraries.
Using mini_portile version 2.0.0
checking for gzdopen() in -lz... yes
checking for iconv... yes
************************************************************************
IMPORTANT NOTICE:

Building Nokogiri with a packaged version of libxml2-2.9.2
with the following patches applied:
	- 0001-Revert-Missing-initialization-for-the-catalog-module.patch
	- 0002-Fix-missing-entities-after-CVE-2014-3660-fix.patch
	- 0003-Stop-parsing-on-entities-boundaries-errors.patch
	- 0004-Cleanup-conditional-section-error-handling.patch
	- 0005-CVE-2015-1819-Enforce-the-reader-to-run-in-constant-.patch
	- 0006-Another-variation-of-overflow-in-Conditional-section.patch
	- 0007-Fix-an-error-in-previous-Conditional-section-patch.patch
	- 0008-CVE-2015-8035-Fix-XZ-compression-support-loop.patch
	- 0009-Updated-config.guess.patch
	- 0010-Fix-parsering-short-unclosed-comment-uninitialized-access.patch

Team Nokogiri will keep on doing their best to provide security
updates in a timely manner, but if this is a concern for you and want
to use the system library instead; abort this installation process and
reinstall nokogiri as follows:

    gem install nokogiri -- --use-system-libraries
        [--with-xml2-config=/path/to/xml2-config]
        [--with-xslt-config=/path/to/xslt-config]

If you are using Bundler, tell it to use the option:

    bundle config build.nokogiri --use-system-libraries
    bundle install

Note, however, that nokogiri is not fully compatible with arbitrary
versions of libxml2 provided by OS/package vendors.
************************************************************************
Extracting libxml2-2.9.2.tar.gz into tmp/x86_64-pc-linux-gnu/ports/libxml2/2.9.2... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxml2/0001-Revert-Missing-initialization-for-the-catalog-module.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxml2/0002-Fix-missing-entities-after-CVE-2014-3660-fix.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxml2/0003-Stop-parsing-on-entities-boundaries-errors.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxml2/0004-Cleanup-conditional-section-error-handling.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxml2/0005-CVE-2015-1819-Enforce-the-reader-to-run-in-constant-.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxml2/0006-Another-variation-of-overflow-in-Conditional-section.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxml2/0007-Fix-an-error-in-previous-Conditional-section-patch.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxml2/0008-CVE-2015-8035-Fix-XZ-compression-support-loop.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxml2/0009-Updated-config.guess.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxml2/0010-Fix-parsering-short-unclosed-comment-uninitialized-access.patch... OK
Running 'configure' for libxml2 2.9.2... OK
Running 'compile' for libxml2 2.9.2... OK
Running 'install' for libxml2 2.9.2... OK
Activating libxml2 2.9.2 (from /usr/local/bundle/gems/nokogiri-1.6.7/ports/x86_64-pc-linux-gnu/libxml2/2.9.2)...
************************************************************************
IMPORTANT NOTICE:

Building Nokogiri with a packaged version of libxslt-1.1.28
with the following patches applied:
	- 0001-Adding-doc-update-related-to-1.1.28.patch
	- 0002-Fix-a-couple-of-places-where-f-printf-parameters-wer.patch
	- 0003-Initialize-pseudo-random-number-generator-with-curre.patch
	- 0004-EXSLT-function-str-replace-is-broken-as-is.patch
	- 0006-Fix-str-padding-to-work-with-UTF-8-strings.patch
	- 0007-Separate-function-for-predicate-matching-in-patterns.patch
	- 0008-Fix-direct-pattern-matching.patch
	- 0009-Fix-certain-patterns-with-predicates.patch
	- 0010-Fix-handling-of-UTF-8-strings-in-EXSLT-crypto-module.patch
	- 0013-Memory-leak-in-xsltCompileIdKeyPattern-error-path.patch
	- 0014-Fix-for-bug-436589.patch
	- 0015-Fix-mkdir-for-mingw.patch
	- 0016-Fix-for-type-confusion-in-preprocessing-attributes.patch
	- 0017-Updated-config.guess.patch

Team Nokogiri will keep on doing their best to provide security
updates in a timely manner, but if this is a concern for you and want
to use the system library instead; abort this installation process and
reinstall nokogiri as follows:

    gem install nokogiri -- --use-system-libraries
        [--with-xml2-config=/path/to/xml2-config]
        [--with-xslt-config=/path/to/xslt-config]

If you are using Bundler, tell it to use the option:

    bundle config build.nokogiri --use-system-libraries
    bundle install
************************************************************************
Extracting libxslt-1.1.28.tar.gz into tmp/x86_64-pc-linux-gnu/ports/libxslt/1.1.28... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0001-Adding-doc-update-related-to-1.1.28.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0002-Fix-a-couple-of-places-where-f-printf-parameters-wer.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0003-Initialize-pseudo-random-number-generator-with-curre.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0004-EXSLT-function-str-replace-is-broken-as-is.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0006-Fix-str-padding-to-work-with-UTF-8-strings.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0007-Separate-function-for-predicate-matching-in-patterns.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0008-Fix-direct-pattern-matching.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0009-Fix-certain-patterns-with-predicates.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0010-Fix-handling-of-UTF-8-strings-in-EXSLT-crypto-module.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0013-Memory-leak-in-xsltCompileIdKeyPattern-error-path.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0014-Fix-for-bug-436589.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0015-Fix-mkdir-for-mingw.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0016-Fix-for-type-confusion-in-preprocessing-attributes.patch... OK
Running git apply with /usr/local/bundle/gems/nokogiri-1.6.7/patches/libxslt/0017-Updated-config.guess.patch... OK
Running 'configure' for libxslt 1.1.28... OK
Running 'compile' for libxslt 1.1.28... OK
Running 'install' for libxslt 1.1.28... OK
Activating libxslt 1.1.28 (from /usr/local/bundle/gems/nokogiri-1.6.7/ports/x86_64-pc-linux-gnu/libxslt/1.1.28)...
checking for main() in -llzma... yes
checking for xmlParseDoc() in libxml/parser.h... yes
checking for xsltParseStylesheetDoc() in libxslt/xslt.h... yes
checking for exsltFuncRegister() in libexslt/exslt.h... yes
checking for xmlHasFeature()... yes
checking for xmlFirstElementChild()... yes
checking for xmlRelaxNGSetParserStructuredErrors()... yes
checking for xmlRelaxNGSetParserStructuredErrors()... yes
checking for xmlRelaxNGSetValidStructuredErrors()... yes
checking for xmlSchemaSetValidStructuredErrors()... yes
checking for xmlSchemaSetParserStructuredErrors()... yes
creating Makefile

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /usr/local/bundle/extensions/x86_64-linux/2.2.0-static/nokogiri-1.6.7/mkmf.log

current directory: /usr/local/bundle/gems/nokogiri-1.6.7/ext/nokogiri
make "DESTDIR=" clean

current directory: /usr/local/bundle/gems/nokogiri-1.6.7/ext/nokogiri
make "DESTDIR="
compiling html_sax_push_parser.c
compiling xml_namespace.c
compiling xml_entity_decl.c
compiling xml_syntax_error.c
compiling xml_xpath_context.c
compiling xml_sax_push_parser.c
compiling xml_document.c
xml_document.c: In function 'set_encoding':
xml_document.c:182:12: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
       free((char *) doc->encoding); /* this may produce a gcc cast warning */
            ^
xml_document.c: In function 'canonicalize':
xml_document.c:537:15: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
       ns[i] = (xmlChar*) ptr;
               ^
compiling xml_sax_parser_context.c
compiling xml_element_decl.c
compiling xml_schema.c
compiling xml_io.c
compiling xml_comment.c
compiling html_sax_parser_context.c
compiling xml_libxml2_hacks.c
compiling xml_processing_instruction.c
compiling xml_attribute_decl.c
compiling xml_element_content.c
compiling xslt_stylesheet.c
In file included from /usr/local/include/ruby-2.2.0/ruby.h:33:0,
                 from ./nokogiri.h:33,
                 from ./xslt_stylesheet.h:4,
                 from xslt_stylesheet.c:1:
xslt_stylesheet.c: In function 'initFunc':
xslt_stylesheet.c:217:21: warning: passing argument 1 of 'rb_data_object_get' makes integer from pointer without a cast
     Data_Get_Struct(ctxt->style->_private, nokogiriXsltStylesheetTuple,
                     ^
/usr/local/include/ruby-2.2.0/ruby/ruby.h:1044:41: note: in definition of macro 'Data_Get_Struct'
     ((sval) = (type*)rb_data_object_get(obj))
                                         ^
/usr/local/include/ruby-2.2.0/ruby/ruby.h:1189:1: note: expected 'VALUE' but argument is of type 'void *'
 rb_data_object_get(VALUE obj)
 ^
xslt_stylesheet.c: In function 'shutdownFunc':
xslt_stylesheet.c:230:21: warning: passing argument 1 of 'rb_data_object_get' makes integer from pointer without a cast
     Data_Get_Struct(ctxt->style->_private, nokogiriXsltStylesheetTuple,
                     ^
/usr/local/include/ruby-2.2.0/ruby/ruby.h:1044:41: note: in definition of macro 'Data_Get_Struct'
     ((sval) = (type*)rb_data_object_get(obj))
                                         ^
/usr/local/include/ruby-2.2.0/ruby/ruby.h:1189:1: note: expected 'VALUE' but argument is of type 'void *'
 rb_data_object_get(VALUE obj)
 ^
compiling html_entity_lookup.c
compiling html_document.c
compiling html_element_description.c
In file included from /usr/local/include/ruby-2.2.0/ruby.h:33:0,
                 from ./nokogiri.h:33,
                 from ./html_element_description.h:4,
                 from html_element_description.c:1:
html_element_description.c: In function 'get_description':
html_element_description.c:252:40: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
   return Data_Wrap_Struct(klass, 0, 0, (void *)description);
                                        ^
/usr/local/include/ruby-2.2.0/ruby/ruby.h:1028:35: note: in definition of macro 'Data_Wrap_Struct'
     rb_data_object_alloc((klass),(sval),(RUBY_DATA_FUNC)(mark),(RUBY_DATA_FUNC)(free))
                                   ^
compiling xml_relax_ng.c
compiling xml_dtd.c
compiling xml_reader.c
compiling xml_cdata.c
compiling xml_entity_reference.c
compiling xml_text.c
compiling xml_attr.c
compiling xml_document_fragment.c
compiling xml_sax_parser.c
compiling xml_encoding_handler.c
compiling xml_node_set.c
xml_node_set.c: In function 'dealloc_namespace':
xml_node_set.c:9:13: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
     xmlFree((xmlChar *)ns->href);
             ^
xml_node_set.c:11:13: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
     xmlFree((xmlChar *)ns->prefix);
             ^
compiling nokogiri.c
compiling xml_node.c
xml_node.c: In function 'reparent_node_with':
xml_node.c:168:7: warning: enumeration value 'XML_ATTRIBUTE_NODE' not handled in switch [-Wswitch]
       switch (reparentee->type) {
       ^
xml_node.c:168:7: warning: enumeration value 'XML_ENTITY_NODE' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_DOCUMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_DOCUMENT_FRAG_NODE' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_NOTATION_NODE' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_HTML_DOCUMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_DTD_NODE' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_ELEMENT_DECL' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_ATTRIBUTE_DECL' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_ENTITY_DECL' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_NAMESPACE_DECL' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_XINCLUDE_START' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_XINCLUDE_END' not handled in switch [-Wswitch]
xml_node.c:168:7: warning: enumeration value 'XML_DOCB_DOCUMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_ATTRIBUTE_NODE' not handled in switch [-Wswitch]
       switch (reparentee->type) {
       ^
xml_node.c:186:7: warning: enumeration value 'XML_ENTITY_NODE' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_DOCUMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_DOCUMENT_TYPE_NODE' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_DOCUMENT_FRAG_NODE' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_NOTATION_NODE' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_HTML_DOCUMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_DTD_NODE' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_ELEMENT_DECL' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_ATTRIBUTE_DECL' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_ENTITY_DECL' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_NAMESPACE_DECL' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_XINCLUDE_START' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_XINCLUDE_END' not handled in switch [-Wswitch]
xml_node.c:186:7: warning: enumeration value 'XML_DOCB_DOCUMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_ELEMENT_NODE' not handled in switch [-Wswitch]
       switch (reparentee->type) {
       ^
xml_node.c:197:7: warning: enumeration value 'XML_ATTRIBUTE_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_CDATA_SECTION_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_ENTITY_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_PI_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_COMMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_DOCUMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_DOCUMENT_TYPE_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_DOCUMENT_FRAG_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_NOTATION_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_HTML_DOCUMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_DTD_NODE' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_ELEMENT_DECL' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_ATTRIBUTE_DECL' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_ENTITY_DECL' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_NAMESPACE_DECL' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_XINCLUDE_START' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_XINCLUDE_END' not handled in switch [-Wswitch]
xml_node.c:197:7: warning: enumeration value 'XML_DOCB_DOCUMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_CDATA_SECTION_NODE' not handled in switch [-Wswitch]
     switch (parent->type) {
     ^
xml_node.c:165:5: warning: enumeration value 'XML_ENTITY_NODE' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_PI_NODE' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_COMMENT_NODE' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_DOCUMENT_TYPE_NODE' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_NOTATION_NODE' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_DTD_NODE' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_ELEMENT_DECL' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_ATTRIBUTE_DECL' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_ENTITY_DECL' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_NAMESPACE_DECL' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_XINCLUDE_START' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_XINCLUDE_END' not handled in switch [-Wswitch]
xml_node.c:165:5: warning: enumeration value 'XML_DOCB_DOCUMENT_NODE' not handled in switch [-Wswitch]
linking shared-object nokogiri/nokogiri.so
Cleaning files only used during build.
rm -rf /usr/local/bundle/gems/nokogiri-1.6.7/ext/nokogiri/tmp/x86_64-pc-linux-gnu/ports
rmdir -p /usr/local/bundle/gems/nokogiri-1.6.7/ext/nokogiri/tmp/x86_64-pc-linux-gnu
rm -rf /usr/local/bundle/gems/nokogiri-1.6.7/ports

No such file or directory - getcwd

Gem files will remain installed in /usr/local/bundle/gems/nokogiri-1.6.7 for inspection.
Results logged to /usr/local/bundle/extensions/x86_64-linux/2.2.0-static/nokogiri-1.6.7/gem_make.out
An error occurred while installing nokogiri (1.6.7), and Bundler cannot
continue.
Make sure that `gem install nokogiri -v '1.6.7'` succeeds before bundling.
The command '/bin/sh -c bundle install' returned a non-zero code: 5
```

The workaround for now is NOT to use the *onbuild* images and apply a bundler config update:

*Dockerfile*
```
FROM ruby:2.2.3
# throw errors if Gemfile has been modified since Gemfile.lock
RUN bundle config --global frozen 1
RUN bundle config build.nokogiri --use-system-libraries
RUN mkdir -p /usr/src/app
WORKDIR /usr/src/app
COPY Gemfile /usr/src/app/
COPY Gemfile.lock /usr/src/app/
RUN bundle install
COPY . /usr/src/app
```

*Docker build then works correctly*
```
Sending build context to Docker daemon 4.096 kB
Step 1 : FROM ruby:2.2.3
 ---> a930a50e60f8
Step 2 : RUN bundle config --global frozen 1
 ---> Using cache
 ---> e2289a8a41ce
Step 3 : RUN bundle config build.nokogiri --use-system-libraries
 ---> Running in 1eca22412ec4
 ---> 5e2cef3115c4
Removing intermediate container 1eca22412ec4
Step 4 : RUN mkdir -p /usr/src/app
 ---> Running in c24f2c5340bf
 ---> e9c2b0df1a28
Removing intermediate container c24f2c5340bf
Step 5 : WORKDIR /usr/src/app
 ---> Running in f816f371a255
 ---> e1ccdc293459
Removing intermediate container f816f371a255
Step 6 : COPY Gemfile /usr/src/app/
 ---> 86ae0599d67e
Removing intermediate container 72d4c24ca95b
Step 7 : COPY Gemfile.lock /usr/src/app/
 ---> 3db140369ff8
Removing intermediate container 573efba21c1c
Step 8 : RUN bundle install
 ---> Running in 77717be179ac
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and
installing your bundle as root will break this application for all non-root
users on this machine.
Fetching gem metadata from https://rubygems.org/.........
Fetching version metadata from https://rubygems.org/..
Installing mini_portile2 2.0.0
Installing nokogiri 1.6.7 with native extensions
Using bundler 1.10.6
Bundle complete! 1 Gemfile dependency, 3 gems now installed.
Bundled gems are installed into /usr/local/bundle.
 ---> 740508d528ac
Removing intermediate container 77717be179ac
Step 9 : ONBUILD copy . /usr/src/app
 ---> Running in 119246adc612
 ---> 96c14833e84e
Removing intermediate container 119246adc612
Successfully built 96c14833e84e
```

**Comment by thaJeztah** [comment]
I know there as issues with overlay in combination with hardlinks an sockets, e.g. https://github.com/docker/docker/issues/12327, https://github.com/docker/docker/issues/12080. Not sure if that's involved here. (One of our maintainers sent a patch to fix that in the Kernel, but unfortunately, it didn't get much  traction yet https://lkml.org/lkml/2015/10/9/620)

**Comment by rared** [comment]
Same problem here.

**Comment by dmcgowan** [comment]
Closing this issue, has been fixed in kernel, see #19082
